### ISSUE TITLE
pcomm process name comm

### ISSUE THREAD

**Issue by brendangregg** [bpftrace/bpftrace#782]
This is to track future work. `comm` returns the thread name (bpf_get_current_comm()), but we may add a bpf_get_current_pcomm() in the future to return the process name. This makes a difference for some Java applications, where `comm` is returning per-thread names, and we'd like a `pcomm` that returned "java".

Until bpf_get_current_pcomm() exists, we could implement a `pcomm` workaround that called curtask and fetched it from the task_struct.

**Comment by danobi** [comment]
Yeah I think this'd be nice. A lot of FB's C++ libraries create their own thread pools and name each thread not very descriptively.

**Comment by cneira** [comment]
@danobi, @brendangregg  is this still needed?

**Comment by danobi** [comment]
Yeah, IMO it'd be nice

**Comment by cneira** [comment]
@danobi,  
I also think the same, I'll work on this.  

Bests

**Comment by cneira** [comment]
@danobi 

I added the change but Andrii suggested another way to do the same without replicating the helper:
> Why not bpf_probe_read_kernel_str(dst, 16, task->real_parent->comm)
> for fentry/fexit/tp_btf/etc BTF-aware BPF programs or
> BPF_CORE_READ_STR_INTO(dst, task, real_parent, comm) for any BPF
> program that has bpf_probe_read_kernel[_str]() (which is pretty much
> every BPF program nowadays, I think)?
>
> Yes, CONFIG_DEBUG_INFO_BTF=y Kconfig is a requirement, but it's a good
> idea to have that if you are using BPF anyways.

Should be close this one?

**Comment by blezsan** [comment]
`curtask->real_parent->comm` doesn't fit the bill exactly, it returns the comm from the parent process which is not necessarily in the same thread group. However `curtask->group_leader->comm` does return the comm for the thread group leader, it's a good workaround for `pcomm` as long as BTF is enabled.

It would be really nice to have `pcomm` be aliased to `curtask->group_leader->comm` under the hood.

**Comment by danobi** [comment]
Would be a good candidate for standard library.

**Comment by jordalgo** [comment]
@Rtoax Just re-reading this issue. It looks like pcomm should be doing `curtask->group_leader->comm`:
- https://github.com/bpftrace/bpftrace/pull/4744
